services/src/main/java/org/keycloak/truststore/SSLSocketFactory.java

```diff
@@ -58,7 +65,17 @@ private SSLSocketFactory() {
             sf = (javax.net.ssl.SSLSocketFactory) javax.net.ssl.SSLSocketFactory.getDefault();
         }
 
-        sslsf = sf;
+        Supplier<String> hostnameProvider = () -> {
+            KeycloakSession session = Resteasy.getProvider().getContextData(KeycloakSession.class);
+            if (session == null) {
+                log.tracef("Not found keycloakSession in the resteasy context when trying to retrieve hostname attribute from it");
+                return null;
+            }
+            String hostname = session.getAttribute(Constants.SSL_SERVER_HOST_ATTR, String.class);
+            log.tracef("Found hostname '%s' to be used by SSLSocketFactory", hostname);
+            return hostname;
+        };
```

Comment:
Since it's only the FIPS1402Provider that needs this, could this be moved to the provider itself and removed from here?

Reply:
The reason is, that this code uses stuff from `keycloak-services`, but `keycloak-crypto-fips1402` doesn't have dependency on `keycloak-services`.

However I think it should work to add dependency on `keycloak-services` into `keycloak-crypto-fips1402` - especially since we have only quarkus distribution and don't need to deal with WF modules etc. So I can change it that way.